src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceCodeRefactoringProvider.State.cs

```diff
@@ -0,0 +1,296 @@
+                if (string.Equals(targetNamespace, declaredNamespace, System.StringComparison.Ordinal))
+                {
+                    return null;
+                }
+
+                // Only provide "move file" action if root namespace is a proper prefix of declared namespace.
+                IsProperPrefix(declaredNamespace, rootNamespace, out var qualifiedIdentifierFromDeclaration);
+                                                                           
+                // Only provide "change namespace" action if it wouldn't cause conflicts in target namespace.
+                if (targetNamespace != null && 
+                    (await WillCauseConflictInTargetNamespace(document.Project, targetNamespace, service.GetDeclaredSymbols(namespaceDeclaration, semanticModel, cancellationToken), cancellationToken).ConfigureAwait(false)))
+                {
+                    targetNamespace = null;
+                }                                                                                                           
+                return new State(document, rootNamespace, qualifiedIdentifierFromFolders, targetNamespace, declaredNamespace, qualifiedIdentifierFromDeclaration);
+            }
+
+
+            private static async Task<bool> WillCauseConflictInTargetNamespace(Project project, string targetNamespace, ImmutableArray<ISymbol> symbols, CancellationToken cancellationToken)
+            {
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                if (!TryGetQualifiedNamespaceWithOutConflict(compilation, targetNamespace, out var namespaceSymbol))
+                {
+                    return true;
+                }
+
+                // Can't cause conflict if target namespace doesn't exist yet.
+                if ((object)namespaceSymbol == null)
+                {
+                    return false;
+                }
+
+                var nameArityMap = new Dictionary<string, List<int>>(StringComparer.Ordinal);
+                foreach (var symbol in symbols)
+                {
+                    if (nameArityMap.TryGetValue(symbol.Name, out var arities))
+                    {
+                        arities.Add(symbol.GetArity());
+                    }
+                    else
+                    {
+                        nameArityMap[symbol.Name] = new List<int>() { symbol.GetArity() };
+                    }
+                }
+
+                foreach (var name in nameArityMap.Keys)
+                {
+                    foreach (var memberWithName in namespaceSymbol.GetMembers(name))
+                    {
+                        if (nameArityMap.TryGetValue(memberWithName.Name, out var arities) && arities.Contains(memberWithName.GetArity()))
+                        {
+                            return true;
+                        }
+                    };
+                }
+
+                return false;
```

Comment:
i'm not sure i understand any of this logic.  If you're moving namespaes around, you can definitely cause conflicts now that your name is in scope where it might not have been.  For example, imagine having a namespace called "Token" that you move into the "Microsoft" namespace.  Now, so many lookups to something called Token may find that name.

Reply:
Could this use the Renamer to prevent any issues?